src/__mocks__/firebase/app.js

```diff
@@ -1,16 +1,26 @@
 var userChangedCb = null;
 
 export const signInCalled = jest.fn();
+import { validEmail, validPassword } from '../../constants/testData';
 
 const authObject = {
   signInWithEmailAndPassword: async (email, pw) => {
     signInCalled();
-    if (email && pw)
+    if (email === validEmail && pw === validPassword)
       return Promise.resolve('Hello');
     else {
-      const error = {
-        code: 'auth/invalid-email'
+      var error = {
+        code: 'auth/unknown'
       };
+      if (email === validEmail && pw === 'invalid') {
```

Comment:
use variable for the invalid pw and on following rows for invalid emails and use same variable in the test file